blocks/rich-text/index.js

```diff
@@ -558,6 +558,38 @@ export class RichText extends Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onChange();
 		}
+
+		// `scrollToCaret` is called on `nodechange`, whereas calling it on
+		// `keyup` *when* moving to a new RichText element results in incorrect
+		// scrolling. Though the following allows false positives, it results
+		// in much smoother scrolling.
+		if ( keyCode !== BACKSPACE && keyCode !== ENTER ) {
+			this.scrollToCaret();
+		}
+	}
+
+	scrollToCaret() {
+		const caretRect = this.getEditorSelectionRect();
+		const caretTop = caretRect.top;
+		if ( caretTop !== this.caretTop ) {
```

Comment:
Still not clear why we need this condition or the `this.caretTop` instance variable.

Reply:
My original thinking was that, when typing in a line, this would be a cheaper test than computing `delta` from `window.pageYOffset`. However, now reading the documentation for `window.scrollY`, of which `pageYOffset` is an alias, there's no mention of any incurred cost of reading this property. In fact, it reads:

> Use this property to check that the document hasn't already been scrolled when using relative scroll functions such as scrollBy(), scrollByLines(), or scrollByPages().

Which is enough to convince me to drop this eq check.